assets/js/theme/custom/mega-menu.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Open mega menu on search button click
+ */
+
+import { constants } from './constants'
+
+export default function megaMenu() {
+    const { MOBILE_WIDTH } = constants
+    const dropdown = document.querySelector('.js-dropdown')
+    const searchButton = document.querySelector('.js-mega-menu-search')
+    const searchInput = document.querySelector('.js-search-quick')
+    const showMoreTriggers = document.querySelectorAll('.js-show-more-item')
+    let hasAttachedEvent = false
```

Comment:
Since the `megaMenu` is called twice, you can't rely on this variable as it will be set to `false` each time the megaMenu is being called. I suggest to move it outside the function, and apply similar solution for the `showMoreMenuItems` - using `data-event-added` isn't a best solution as you may need to attach multiple events, and in such case the `data-event-added` won't have any meaning.

Reply:
I've updated it. Please check again.  By using button.dataset.hasShowMoreEvent = 'true', I can track each button's event listener attachment directly on the element itself. This ensures that we don't unintentionally block event listeners from being added to other "Show More" buttons while still preventing duplicates on the same button.

Reply:
Is it possible to not attach an event to all buttons if you are calling `addEventListener` inside a loop?